File: Medium/2785. Sort Vowels in a String/2785. Sort Vowels in a String.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowel_sorted = [char for char in s if char.lower() in "aeiou"]
        vowel_sorted.sort(reverse=True)

        result = ""
        for char in s:
            if char.lower() in "aeiou":
                result += vowel_sorted.pop()
            else:
                result += char

        return result

File: Medium/2785. Sort Vowels in a String/2785. Sort Vowels in a String.ts
function sortVowels(s: string): string {
    let vowelSorted = [];

    for (let char of s) {
        if ("aeiouAEIOU".includes(char)) {
            vowelSorted.push(char);
        }
    }

    vowelSorted.sort().reverse();

    let res = "";
    for (let char of s) {
        if ("aeiouAEIOU".includes(char)) {
            res += vowelSorted.pop();
        } else {
            res += char;
        }
    }

    return res;
}
